### ISSUE TITLE
In the classification of the 3classes, the accuracy is not more than 0.46 

### ISSUE THREAD

**Issue by smnfouladi** [boncey/Flickr4Java#353]
i have a network for classify data into 3 class.data are 13760 Two-dimensional array with 19 row and 1280 column. first trained network by convolution autoencoder and save model.h5 then get encoder output and re-train in multi layer and finally use for classification. model accuracy is always 0.46. i change lots of parameters,change learning rate,loss function,optimizer,deep of network,use different layer,normalize data,....but the accuracy didn't improve even convert conv1d to con2d but result is same. I don't know what else to do?

Below is the code for the classification section: first load modeland then train encoder and classify 
 
    autoencoder.load_weights("/content/model.h5")
    print("Loaded Model from disk")

    weights_list = autoencoder.get_weights()

    autoencoder.layers.pop()
    autoencoder.layers.pop()
    autoencoder.layers.pop()
    autoencoder.layers.pop()
    autoencoder.layers.pop()
    autoencoder.layers.pop()
    autoencoder.layers.pop()
    autoencoder.layers.pop()
    autoencoder.layers.pop()
    autoencoder.layers.pop()
    autoencoder.layers.pop()

    autoencoder.compile(Adam(lr=0.001, beta_1=0.9, beta_2=0.999,      epsilon=None, decay=0.0,  amsgrad=False), loss='categorical_crossentropy')

    autoencoder.summary()

    #classification using encoded output

    y=Conv1D(128, 4, activation='relu', padding='same')(encoded)
    y=Conv1D(128,4,padding='same',activation='relu')(y)
    y=BatchNormalization()(y)
    y = MaxPooling1D(2, name='max722')(y)    
    y=Dropout(0.25)(y)

    y=Conv1D(64, 4, activation='relu', padding='same', name='conv3238')(y)
    y=Conv1D(64,4,padding='same',activation='relu')(y)
    y=BatchNormalization()(y)
    y = MaxPooling1D(2, name='max788')(y)    
    y=Dropout(0.25)(y)

    y=Conv1D(32, 4, activation='relu', padding='same', name='conv8')(y)
    y=Conv1D(32,4,padding='same',activation='relu')(y)
    y=BatchNormalization()(y)
    y = MaxPooling1D(2, name='max7')(y)    
    y=Dropout(0.25)(y) 

    y = Conv1D(16, 4, padding='same', activation='relu', name = 'conv9')(y)
    y = Conv1D(16, 4, padding='same',activation='relu')(y)
    y = BatchNormalization()(y)
    y = MaxPooling1D(2, name='max8')(y)
    y = Dropout(0.25)(y)  

    y = Conv1D(8, 4, padding='same', activation='relu')(y)
    y = Conv1D(8, 4, padding='same',activation='relu')(y)
    y = BatchNormalization()(y)
    y = MaxPooling1D(2)(y)
    y = Dropout(0.25)(y) 

    y=Flatten()(y)

    y = Dense(256,  activation='relu',kernel_regularizer=regularizers.l2(0.01),
            activity_regularizer=regularizers.l1(0.01))(y)

    y = Dropout(0,5)(y)     
    y1 = Dense(3, activation='softmax',kernel_regularizer=regularizers.l2(0.01),
            activity_regularizer=regularizers.l1(0.01))(y)     
    classifier= Model(inputs=autoencoder.input, outputs=y1)
    classifier.summary()

     arrayofdata=[] 
     inputdata = '/content/Cleaned'
     countfile = len([name for name in os.listdir(inputdata) if   os.path.isfile(os.path.join(inputdata, name))])
     files = os.listdir(inputdata)
     for filename in glob.glob(os.path.join(inputdata, '*.mat')):

      mat = sio.loadmat(filename, squeeze_me=True, struct_as_record=False)     
      inputdata = mat['Data']
      inputdata_ = np.hsplit(inputdata, [1280])
      inputdata = inputdata_[0]        
      arrayofdata.append(inputdata.tolist())
      arrayofdata_=arrayofdata     
      arrayofdata_ = np.array(arrayofdata_)

       arrayoflable=[] 
       inputdata1 = '/content/Target'
        countfile2 = len([name for name in os.listdir(inputdata1) if os.path.isfile(os.path.join(inputdata1, name))])
     files1 = os.listdir(inputdata1)
     for filename1 in glob.glob(os.path.join(inputdata1, '*.mat')):

     mat1 = sio.loadmat(filename1, squeeze_me=True, struct_as_record=False)     
    inputdata1 = mat1['Data']
    inputdata1 = inputdata1[0]
    arrayoflable.append(inputdata1.tolist())
    arrayoflable_=arrayoflable

    labels = np.array(arrayoflable_)    
    one_hot_labels = to_categorical(labels, num_classes=3)    
    x_train, x_test, y_train, y_test = train_test_split(arrayofdata_,
                                                      one_hot_labels,
                                                      test_size=0.33,
                                                      random_state=42)

    x_train = x_train.astype('float32') / 255
    x_test = x_test.astype('float32') / 255
    x_train = np.reshape(x_train, (x_train.shape[0],x_train.shape[1]*x_train.shape[2],1))
    x_test= np.reshape(x_test, (x_test.shape[0],x_test.shape[1]*x_test.shape[2],1))

    amin=min(x_train.flatten())
    amax=max(x_train.flatten())

     for i, val in enumerate(x_train):   
     x_train[i] = (val-amin) / (amax-amin)

    amintest=min(x_test.flatten())
    amaxtest=max(x_test.flatten())

    for i, val in enumerate(x_test):  
      x_test[i] = (val-amaxtest) / (amaxtest-amintest) 

    from sklearn.utils import compute_class_weight
    w = compute_class_weight('balanced', np.unique(labels), labels)
    weights = {
     np.unique(labels)[0] : w[0], 
     np.unique(labels)[1] : w[1],
     np.unique(labels)[2] : w[2] }
  

    classifier.compile(SGD(lr=0.001),loss='categorical_crossentropy', metrics=['categorical_accuracy'])

     history= classifier.fit(x_train,y_train,epochs=5,batch_size=16,
                            shuffle=True,  
                            verbose=0,class_weight=weights,                         
                            validation_data=(x_test, y_test),
                            callbacks=[TensorBoard(log_dir='/content/tmp/autoencoder')])

     print('val_loss:',mean(history.history['val_loss']))
     print('loss:',mean(history.history['loss']))         
     print('val_categorical_accuracy:',mean(history.history['val_binary_accuracy'])) 
    print('categorical_accuracy:',mean(history.history['binary_accuracy']))
